controllers/component_dependency_update_controller.go

```diff
@@ -339,23 +339,42 @@ func (r *ComponentDependencyUpdateReconciler) handleCompletedBuild(ctx context.C
 	}
 
 	for i := range components.Items {
-		comp := components.Items[i]
-		if slices.Contains(updatedComponent.Spec.BuildNudgesRef, comp.Name) {
-			toUpdate = append(toUpdate, comp)
+		component := components.Items[i]
+		if slices.Contains(updatedComponent.Spec.BuildNudgesRef, component.Name) {
+			gitProvider, err := getGitProvider(component)
```

Comment:
`getGitProvider` is and expensive operation. Can we reduce number of its calls in a reasonable way?